src/Utils/Library.dfy

```diff
@@ -287,6 +292,26 @@ module Utils.Lib.Seq {
             case Some(y) => [y]
             case None => []) + MapFilter(s[1..], f)
   }
+
+  function method Break<T>(f: T -> bool, s: seq<T>): (result: (seq<T>, seq<T>))
+    ensures |result.1| <= |s|
+    ensures forall c | c in result.0 :: !f(c)
+  {
+    if |s| == 0 then ([], [])
+    else if f(s[0]) then ([], s)
+    else
+      var (h, t) := Break(f, s[1..]);
+      ([s[0]] + h, t)
+  }
+
+  function method Split<T(==)>(x: T, s: seq<T>): (result: seq<seq<T>>)
+    decreases |s|
+    ensures forall s | s in result :: x !in s
```

Comment:
Can we prove that concatenating all the resulting sequences returns the original sequence, filtered?

Reply:
Probably, though it's not an automatic proof. I've added a comment about it, should we care to prove it in the future. But I'm not sure it's high priority to prove it right now.